integration/e2e_test.go

```diff
@@ -302,11 +321,38 @@ func deployMicrobot(t testing.TB) {
 
 	// Make sure we have as many nodes as machines
 	attempt := 0
-	maxAttempts := 60
+	maxAttempts := 120
 	t.Log("Waiting for the deployment to complete")
 	command = []string{"kubectl", "--kubeconfig=" + KUBECONFIG, "wait", "deploy/bot", "--for=jsonpath={.status.readyReplicas}=30"}
 	for {
 		cmd = exec.Command(command[0], command[1:]...)
+		outputBytes, err = cmd.CombinedOutput()
+		if err != nil {
+			t.Log(string(outputBytes))
+			if attempt >= maxAttempts {
+				t.Fatal(err)
+			} else {
+				attempt++
+				t.Log("Retrying")
+				time.Sleep(10 * time.Second)
+			}
+		} else {
+			break
+		}
+	}
+}
+
+// validateNoCalico checks a there is no calico daemon set.
+func validateNoCalico(t testing.TB) {
+	t.Log("Validate no Calico daemon set")
+
+	t.Log("Checking for calico daemon set")
+	command := []string{"kubectl", "--kubeconfig=" + KUBECONFIG, "-n", "kube-system", "get", "ds"}
```

Comment:
any reason why you not directly try to get the calico daemonset and retry if this fails?

Reply:
in some cases cluster is not ready and may return error related to kubectl connecting to the cluster

Reply:
but that is fine, no?
You just retry the `kubectl get ds calico-...` and until either the max attempts are reached or `err == nil`

Reply:
but it there is calico ds it will return even if we don't want to have it